crates/assistant_context_editor/src/context_store.rs

```diff
@@ -336,18 +335,17 @@ impl ContextStore {
                 self.synchronize_contexts(cx);
             }
             project::Event::DisconnectedFromHost => {
-                self.contexts.retain_mut(|context| {
-                    if let Some(strong_context) = context.upgrade() {
```

Comment:
Do you know why there are all of these `if let Some()`s being replaced by `match`es?

Reply:
https://doc.rust-lang.org/edition-guide/rust-2024/temporary-if-let-scope.html

drop order of if let was changed, so this is only safe migration. 

mostly these are likely revertable

Reply:
I think that's due to changes around `if let` bindings lifetimes: https://doc.rust-lang.org/nightly/edition-guide/rust-2024/temporary-if-let-scope.html

Honestly though, most of these seem benign - thus I'm working on migrating us to 2024 edition manually, as I don't think it's worth it to change all if lets to matches.